src/pages/Expenses.py

```diff
@@ -597,37 +607,32 @@ def init(conn, conn_changes, data_dir, user):
                 add_row(df=df_initial, conn=conn_changes)
 
             df = run_sql(user_input, df_initial, table_name="expenses")
-            df = df.round(2)
-            gb = GridOptionsBuilder.from_dataframe(df)
-            gridOptions = gb.build()
-            gridOptions["editable"] = True
-            gridOptions["sortable"] = True
-            gridOptions["filter"] = True
-            gridOptions["columnDefs"] = [
-                {
-                    "headerName": col,
-                    "field": col,
-                    "editable": col not in {"pk", "source_file", "line"},
-                }
-                for col in df.columns
-            ]
-            edits = AgGrid(
-                df,
-                gridOptions,
-                data_return_mode="AS_INPUT",
-                update_mode="MODEL_CHANGED",
-                fit_columns_on_grid_load=False,
-                theme="streamlit",  # Add theme color to the table
-                enable_enterprise_modules=True,
-                height=350,
-                width="100%",
-                reload_data=False,
+            df_table = df.round(2)
+            df_table["date"] = df_table["date"].apply(lambda x: parse(x))
+            st.data_editor(
```

Comment:
btw @mukundvarma  , this is the part that switches from using the aggrid table to the new streamlit data editor table.